### PR TITLE
[zh-cn]: update the translations of Atomics notify(), wait() and waitAsync()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#17175

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md [hunk 1]
```diff
@@ -1,55 +1,63 @@
 ---
 title: Atomics.notify()
 slug: Web/JavaScript/Reference/Global_Objects/Atomics/notify
+l10:
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md:4]:
```suggestion
l10n:
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md [hunk 2]
```diff
@@ -1,55 +1,63 @@
...
 ---
 
 {{JSRef}}
 
-静态方法 **`Atomics.notify()`** 提醒一些在等待队列中休眠的代理。
+**`Atomics.notify()`** 静态方法唤醒一些在等待队列中休眠的代理。
 
-> **备注：** 本操作仅在共享的 {{jsxref("Int32Array")}} 下可用。
+> **备注：** 此操作仅适用于查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
+> 对于非共享的 `ArrayBuffer` 对象，其将返回 `0`。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md:13]: How about "基于 xxx 的 xxx 视图"

```suggestion
> **备注：** 此操作仅适用于基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}} 视图。对于非共享的 `ArrayBuffer` 对象，其将返回 `0`。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md [hunk 3]
```diff
@@ -1,55 +1,63 @@
...
-```plain
+```js-nolint
 Atomics.notify(typedArray, index, count)
 ```
 
 ### 参数
 
 - `typedArray`
-  - : 一个共享的 {{jsxref("Int32Array")}}。
+  - : 查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md:24]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md [hunk 4]
```diff
@@ -1,55 +1,63 @@
...
-被唤醒的代理的数量。
+- 返回唤醒的代理数量。
+- 如果在非共享的 {{jsxref("ArrayBuffer")}} 上调用则返回 `0`。
 
 ### 异常
 
-- 若 `typedArray` 不是共享的 {{jsxref("Int32Array")}}，则抛出一个 {{jsxref("TypeError")}} 异常。
-- 若 `index` 索引超出了 `typedArray` 的大小，则抛出一个 {{jsxref("RangeError")}} 异常。
+- {{jsxref("TypeError")}}
+  - : 如果 `typedArray` 不是一个查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}，则抛出该异常。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md:38]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md [hunk 5]
```diff
@@ -1,55 +1,63 @@
...
 
 ```js
-var sab = new SharedArrayBuffer(1024);
-var int32 = new Int32Array(sab);
+const sab = new SharedArrayBuffer(1024);
+const int32 = new Int32Array(sab);
 ```
 
-一个读线程会进入休眠并监视索引 0 处的值（默认为 0）。只要索引 0 处的值不为 0，读进程就会唤醒。但是，一旦写进程存储了一个新的值，写进程就会产生一个提醒并返回写入后的新值（123）。（这里示例有问题或者说对初学者不友好，如果直接在浏览器控制台运行下面代码会报错，因为我们不能尝试睡眠主线程，可以见[重学 js——结构化数据之 Atomics 对象](https://github.com/lizhongzhen11/lizz-blog/issues/125#notice)，同时我在 **codepen** 写了一个示例：[Atomics.wait 使用示例](https://codepen.io/lizhongzhen11/project/editor/AmzyaY#)）
+令一个读取线程休眠并等待位置 0，其预期为 0。只要条件一直为真，则将不会继续运行。然而，一旦写入线程存储了一个新的值，它将被读取线程唤醒并返回新的值（123）。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md:53]:
```suggestion
令一个读取线程休眠并在位置 0 处等待，预期该位置的值为 0。只要条件一直为真，则将不会继续运行。然而，一旦写入线程存储了一个新的值，它将被读取线程唤醒并返回新的值（123）。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md [hunk 6]
```diff
@@ -1,57 +1,67 @@
...
 ---
 
 {{JSRef}}
 
-静态方法 **`Atomics.wait()`** 确保了一个在 {{jsxref("Int32Array")}} 数组中给定位置的值没有发生变化、仍然是给定的值时进程将会睡眠，直到被唤醒或超时。该方法返回一个字符串，值为`"ok"`, `"not-equal"`, 或 `"timed-out"` 之一。
+**`Atomics.wait()`** 静态方法验证共享内存特定位置是否仍然包含给定值，如果是则休眠，直到被唤醒或超时。其返回一个内容为 `"ok"`、`"not-equal"` 或 `"timed-out"` 的字符串。
 
-> **备注：** 这项操作仅允许同一个共享内存的 {{jsxref("Int32Array")}} 配合使用并且无法运行在主线程中。
+> **备注：** 此操作仅适用于查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}} 并且在主线程中可能不可用。
+> 有关此方法的非阻塞异步版本，请参见 {{jsxref("Atomics.waitAsync()")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md:13]: combine those two line

File: files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md [hunk 7]
```diff
@@ -1,57 +1,67 @@
...
+```js-nolint
+Atomics.wait(typedArray, index, value)
+Atomics.wait(typedArray, index, value, timeout)
 ```
 
 ### 参数
 
 - `typedArray`
-  - : 一个共享内存的 {{jsxref("Int32Array")}} 数组。
+  - : 查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md:25]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md [hunk 8]
```diff
@@ -1,57 +1,67 @@
...
-一个 {{jsxref("String")}} 字符串，值为 "`ok`", "`not-equal`", 或 "`timed-out`" 三种之一。
+一个内容为 `"ok"`、`"not-equal"` 或 `"timed-out"` 的字符串。
 
 ### 异常
 
-- 如果参数 `typedArray` 不是一个共享内存的 {{jsxref("Int32Array")}} 数组，将会抛出一个 {{jsxref("TypeError")}} 。
-- 如果参数 `index` 超出了参数 `typedArray`的边界，将会抛出一个 {{jsxref("RangeError")}} 。
+- {{jsxref("TypeError")}}
+  - : 有下列情况之一，则抛出该异常：
+    - 如果 `typedArray` 不是一个查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md:41]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md [hunk 9]
```diff
@@ -1,57 +1,67 @@
...
 
 ```js
-var sab = new SharedArrayBuffer(1024);
-var int32 = new Int32Array(sab);
+const sab = new SharedArrayBuffer(1024);
+const int32 = new Int32Array(sab);
 ```
 
-检测给定的数组索引 0 的值，如果它如预期一般的等于我们给定的值 0，则这个读取线程将会睡眠等待。一旦当有一个写入线程在这个位置存储了一个新值，它将会收到写入线程的通知并且返回新值 (123) :
+令一个读取线程休眠并等待位置 0，其预期为 0。只要条件一直为真，则将不会继续运行。然而，一旦写入线程存储了一个新的值，它将被读取线程唤醒并返回新的值（123）。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md:57]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  sourceCommit: 27180875516cc311342e74b596bfb589b7211e0c
+---
+
+{{JSRef}}
+
+**`Atomics.waitAsync()`** 静态方法异步等待共享内存的特定位置并返回一个 {{jsxref("Promise")}}。
+
+与 {{jsxref("Atomics.wait()")}} 不同，`waitAsync` 是非阻塞的且可用于主线程。
+
+> **备注：** 此操作仅适用于查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:14]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+```js-nolint
+Atomics.waitAsync(typedArray, index, value)
+Atomics.waitAsync(typedArray, index, value, timeout)
+```
+
+### 参数
+
+- `typedArray`
+  - : 查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:26]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+- `async`
+  - : 一个布尔值，指示 `value` 属性是否为 {{jsxref("Promise")}}。
+- `value`
+  - : 如果 `async` 是 `false`，它将是一个内容为 `"not-equal"` 或 `"timed-out"` 的字符串（仅当 `timeout` 参数为 `0` 时）。如果 `async` 是 `true`，它将会是一个 {{jsxref("Promise")}}，其兑现值为一个内容为 `"ok"` 或 `"timed-out"` 的字符串。这个 promise 永远不会被拒绝。
+
+### 异常
+
+- {{jsxref("TypeError")}}
+  - : 如果 `typedArray` 不是一个查看 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}，则抛出该异常。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:46]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+### 使用 waitAsync()
+
+给定一个共享的 `Int32Array`。
+
+```js
+const sab = new SharedArrayBuffer(1024);
+const int32 = new Int32Array(sab);
+```
+
+令一个读取线程休眠并等待位置 0，其预期为 0。`result.value` 将是一个 promise。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:61]: +1

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+```js
+const result = Atomics.waitAsync(int32, 0, 0, 1000);
+// { async: true, value: Promise {<pending>} }
+```
+
+在该读取线程或另一个线程中，在内存位置 0 调用以令该 promise 解决为 `"ok"`。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:68]: minor fixes

```suggestion
在该读取线程或另一个线程中，对内存位置 0 调用以令该 promise 解决为 `"ok"`。
```

**Review by yin1999 — COMMENTED** [review]
Some points to consider:

File: files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md (hunk 3 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md:24]:
```suggestion
  - : 基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md (hunk 4 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/notify/index.md:38]:
```suggestion
  - : 如果 `typedArray` 不是一个基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}，则抛出该异常。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md (hunk 6 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md:13]:
```suggestion
> **备注：** 此操作仅适用于基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}} 并且在主线程中可能不可用。有关此方法的非阻塞异步版本，请参见 {{jsxref("Atomics.waitAsync()")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md (hunk 7 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md:25]:
```suggestion
  - : 基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md (hunk 8 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md:41]:
```suggestion
    - 如果 `typedArray` 不是一个基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md (hunk 10 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:14]:
```suggestion
> **备注：** 此操作仅适用于基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md (hunk 11 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:26]:
```suggestion
  - : 基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md (hunk 12 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:46]:
```suggestion
  - : 如果 `typedArray` 不是一个基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}，则抛出该异常。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md [hunk 15]
```diff
@@ -1,57 +1,66 @@
...
+  sourceCommit: 27180875516cc311342e74b596bfb589b7211e0c
 ---
 
 {{JSRef}}
 
-静态方法 **`Atomics.wait()`** 确保了一个在 {{jsxref("Int32Array")}} 数组中给定位置的值没有发生变化、仍然是给定的值时进程将会睡眠，直到被唤醒或超时。该方法返回一个字符串，值为`"ok"`, `"not-equal"`, 或 `"timed-out"` 之一。
+**`Atomics.wait()`** 静态方法验证共享内存特定位置是否仍然包含给定值，如果是则休眠，直到被唤醒或超时。其返回一个内容为 `"ok"`、`"not-equal"` 或 `"timed-out"` 的字符串。
 
-> **备注：** 这项操作仅允许同一个共享内存的 {{jsxref("Int32Array")}} 配合使用并且无法运行在主线程中。
+> **备注：** 此操作仅适用于基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}} 并且在主线程中可能不可用。有关此方法的非阻塞异步版本，请参见 {{jsxref("Atomics.waitAsync()")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/wait/index.md:12]:
```suggestion
> **备注：** 此操作仅适用于基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}} 视图，并且在主线程中可能不可用。有关此方法的非阻塞异步版本，请参见 {{jsxref("Atomics.waitAsync()")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+  sourceCommit: 27180875516cc311342e74b596bfb589b7211e0c
+---
+
+{{JSRef}}
+
+**`Atomics.waitAsync()`** 静态方法异步等待共享内存的特定位置并返回一个 {{jsxref("Promise")}}。
+
+与 {{jsxref("Atomics.wait()")}} 不同，`waitAsync` 是非阻塞的且可用于主线程。
+
+> **备注：** 此操作仅适用于基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/atomics/waitasync/index.md:14]:
```suggestion
> **备注：** 此操作仅适用于基于 {{jsxref("SharedArrayBuffer")}} 的 {{jsxref("Int32Array")}} 或 {{jsxref("BigInt64Array")}} 视图。
```

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t

**Pull request merged**